doc/source/howtos/upgrade_guide.rst

```diff
@@ -117,3 +120,41 @@ To upgrade the cluster, follow these steps for each node:
 #. Repeat this procedure for the next node in the cluster
    until you upgrade all nodes.
 
+It is important that when rejoining, the node should synchronize using
+:term:`IST`.
+
+When performing any upgrade (major or minor), :term:`SST` could
+be initiated by the joiner node after the upgrade if the server
+was offline for some time. After :term:`SST` completes, the data
+directory structure needs to be upgraded (using mysql_upgrade)
+once more time to ensure compatibility with the newer version
+of binaries.
+
+   .. note:: In case of :term:`SST` synchronization,  messages log
+             file contains statements like "Check if state gap can
+             be serviced using IST ... State gap can't be serviced
+             using IST. Switching to SST" instead of "Receiving
+             IST: ..." lines appropriate to  :term:`IST`
+             synchronization.
+
+
+To upgrade data directory structure after :term:`SST`, following
+additional steps should be done:
+
+11. shutdown node that rejoined cluster using :term:`SST`:
+
+    .. code-block:: bash
+
+       $ sudo service mysql stop
+
+#. restart the node in standalone mode by setting the
+   :variable:`wsrep_provider` variable to ``none``, e.g.:
+
+   .. code-block:: bash
+
+      sudo mysqld --skip-grant-tables --user=mysql --wsrep-provider='none'
+
+#. run ``mysql-upgrade``
+
+#. restart the node in cluster mode (e.g by executing ``sudo service mysql
+   start`` and make sure the cluster joins back using :term:`IST`.
```

Comment:
s/and make sure the cluster joins back using/ensuring that the node rejoins the cluster using/

Reply:
Hm... as far as I understand, it is still possible that node will continue to attempt SST rejoins.  "ensuring" may be understood as related to "sudo service mysql start", which looks to be not correct... This "make sure" appeals to previous comment (lines in the error log). If speaking about real instruction to "ensure", I've found only low-level manual operations (file copy, etc) to avoid SST, so perhaps it's not a good place for such details...

Reply:
Perhaps something like 
s/and make sure the cluster joins back using :term:`IST`./and check if the node rejoins the cluster using :term:`IST`. If the node rejoins the cluster using SST, please refer to the SST related instructions above./